crates/common/src/config.rs

```diff
@@ -34,73 +67,214 @@ pub fn safe_default<V: Debug + Eq>(opt: Option<V>, default: V) -> anyhow::Result
     }
 }
 
+/// Computes the hash of the genesis system configuration.
+///
+/// # Arguments
+///
+/// * `system_config` - A reference to a `SystemConfig` struct containing all the necessary
+///   configuration fields to generate the hash.
+///
+/// # Returns
+///
+/// This function returns a `Result` containing a 32-byte array representing the hash of the
+/// system configuration. In case of an error (e.g., unsafe defaults or conversion failures),
+/// it returns an error wrapped in `anyhow::Error`.
+///
+/// # Algorithm
+///
+/// This function computes the hash in the following steps:
+/// 1. Extracts individual fields from the `system_config` and converts them into byte slices:
+///    - `batcher_address`: Concatenates the address as a byte slice.
+///    - `overhead`, `scalar`, and `gas_limit`: Converts each to 32-byte big-endian representations.
+///    - Defaulted fields (`base_fee_scalar`, `blob_base_fee_scalar`, `eip1559_denominator`,
+///      `eip1559_elasticity`, `operator_fee_scalar`, and `operator_fee_constant`): Each field is
+///      converted to its big-endian byte representation, using safe defaults when necessary. If a
+///      default fails, the function propagates the error context.
+///
+/// 2. Concatenates all the byte slice representations of the fields into a single buffer.
+///
+/// 3. Computes a cryptographic hash of the concatenated buffer using the `SHA2` hashing
+///    algorithm.
+///
+/// 4. Converts the resulting hash into a fixed-size 32-byte array.
+///
+/// # Errors
+///
+/// This function may fail in the following scenarios:
+/// - If the `safe_default` function for any of the defaulted fields fails to produce a valid
+///   value, an error will be returned with additional context.
+///
+/// # Notes
+///
+/// - The hash is computed deterministically based on the input `SystemConfig`. Any changes to
+///   the configuration will result in a different hash.
+/// - It is important to ensure that the input fields adhere to the expected formats and ranges
+///   for proper hash computation.
+pub fn genesis_system_config_hash(system_config: &SystemConfig) -> anyhow::Result<[u8; 32]> {
+    let fields = [
```

Comment:
There are encodings specifically developed in the Blockchain context to lead to deterministic and unique encodings which can be used for hashing. An example here would be CBOR, which could be used as an alternative to that manual approach.